linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -482,17 +501,36 @@ final class Emitter(config: Emitter.Config, prePrinter: Emitter.PrePrinter) {
       }
     }
 
-    val isJSClass = kind.isJSClass
-
     // Class definition
-    if (linkedClass.hasInstances && kind.isAnyNonNativeClass) {
-      val isJSClassVersion = Version.fromByte(if (isJSClass) 1 else 0)
+    if (uncachedDecisions.hasInstances && kind.isAnyNonNativeClass) {
+      /* Decision-making in this scope is governed by the `fullClassChangeTracker`
+       * below instead of the uncachedDecisions. It is actually called at the
+       * end of the block, because it needs to track *produced* js.Trees in
+       * addition to *inputs* in the LinkedClass.
+       *
+       * Nevertheless, since it tracks the class version, we are allowed to
+       * use the following facts for uncached decision-making.
+       */
+
+      val hasJSSuperClass = linkedClass_!.jsSuperClass.isDefined
+
+      /* The optimizer cannot remove the class initializer, so this is covered
+       * by the class version as well.
```

Comment:
- It's not clear to me why addition or removal of a class initializer must trigger a class version bump. At least taken at face value, it's just a method, and we version these separately.
- Reliance on the optimizer's behavior is quite brittle IMO.

Are there major downsides to just putting this into `UncachedDecisions`?

Reply:
The comment is clear now. IIUC, proposals like https://github.com/scala-js/scala-js/issues/1593 could break this.

However, let's maybe not block this PR on this: it clearly improves that status quo and it seems this can be easily fixed later.